packages/tenant-discovery-react/src/plugin.tsx

```diff
@@ -0,0 +1,235 @@
+                const formFields = input.formFields;
+
+                if (formFields.length < 1) {
+                  // Very unlikely since we are setting the field above
+                  // but we should still handle.
+                  throw new Error("Should never come here");
+                }
+                const tenantValueEntered = formFields[0]?.value;
+
+                // Set the tenantId in the current URL and refresh the page
+                implementation.setTenantId(tenantId, tenantValueEntered);
+
+                // return a SIGN_IN_NOT_ALLOWED error
+                // if the code reached this point.
+                // though it should not since we refresh the page right after the tenant
+                // is set.
+                return {
+                  status: "SIGN_IN_NOT_ALLOWED",
+                  fetchResponse: new Response(),
+                  reason: "Tenant discovery plugin overridden method",
+                };
+              },
+            };
+          },
+        },
+      },
+      generalAuthRecipeComponentOverrides: {
+        AuthPageHeader_Override: ({ DefaultComponent, ...props }) => {
+          const { shouldShowSelector } = parseTenantId();
+
+          if (shouldShowSelector) {
+            // eslint-disable-next-line react/jsx-no-literals
+            return <div>Enter email to continue</div>;
+          }
+
+          // @ts-ignore
+          return <DefaultComponent {...props} />;
+        },
+        AuthPageComponentList_Override: ({ DefaultComponent, ...props }) => {
+          const { shouldShowSelector } = parseTenantId();
+
+          // Update the auth component list to only keep email if we are showing
+          // the selector.
+          if (shouldShowSelector) {
+            const emailPasswordSignInComponent = EmailPasswordPreBuiltUI.getInstanceOrInitAndGetInstance()
+              .getAuthComponents()
+              .find((comp) => comp.type === "SIGN_IN");
+
+            // Unlikely that component will be undefined but still handle
+            // the case.
+            if (!emailPasswordSignInComponent) {
+              throw new Error("Should never come here");
+            }
+
+            // Fix type error by ensuring the component matches expected type
+            props.authComponents = [emailPasswordSignInComponent.component as React.FC<AuthComponentProps>];
+
+            // Set the formField to hide the password field
+            props.factorIds = ["emailpassword"];
+
+            props.userContext = {
+              ...props.userContext,
+              shouldShowSelector: true,
+            };
+          }
+
+          useLayoutEffect(() => {
+            if (shouldShowSelector) {
+              hidePasswordInput();
+              updateSignInSubmitBtn(translations("PL_TD_EMAIL_DISCOVERY_CONTINUE_BUTTON"));
+            }
+
+            // Try to populate the email if it is present
+            // in the URL.
+            populateEmailFromUrl();
```

Comment:
We may have a way to avoid a full page redirect, in which case we could avoid storing and repopulating the email field. Let's discuss this tomorrow.